### PR TITLE
Remove generate-tarball go dependencies

### PR DESCRIPTION
**Pull request by tieujason330**
- @Xenorith @jiacheliu3 
- Replace cmdline vendor pkg with go's `flag` package
- requested by @jiacheliu3

### CODE REVIEW

**Comment by tieujason330** [comment]
alluxio-bot, merge this please

File: dev/scripts/src/alluxio.org/build-distribution/main.go [hunk 1]
```diff
@@ -12,11 +12,35 @@
...
+
+func runSubcmd(args []string) error {
+	subcmdNames := []string{generateTarball, generateReleaseTarball}
+	if len(args) < 2 {
+		return fmt.Errorf("expected a subcommand in arguments. use one of %v", subcmdNames)
+	}
+
+	if args[1] == generateTarball {
+		return cmd.Single(args)
+	} else if args[1] == generateReleaseTarball {
```
Comment (ZacBlanco) [inline dev/scripts/src/alluxio.org/build-distribution/main.go]: maybe pull `args[1]` into a variable? Otherwise lgtm

Comment (tieujason330) [reply dev/scripts/src/alluxio.org/build-distribution/main.go]: updated

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Pull request merged**